src/tensors/allocator.h

```diff
@@ -122,13 +122,21 @@ class Allocator {
     size = align(size);
     auto it = std::lower_bound(gaps_.begin(), gaps_.end(), Gap(nullptr, size));
 
-    if(throw_ && it == gaps_.end()) {
-      throw AllocationException(available_, size);
-    }
+    if (it == gaps_.end()) {
+      if (throw_) {
+        throw AllocationException(available_, size);
+      }
+
+      size_t last_gap_size = 0;
```

Comment:
Please add comments what exactly this is doing and how this is different to before. It took me a while to figure this out. From what I am seeing it either grows step size or required minimal size, depending what is larger. 

Otherwise a good change probably.